march_joint_inertia_controller/src/inertia_estimator.cpp

```diff
@@ -0,0 +1,290 @@
+
+  filtered_joint_torque_.resize(fil_tor_size_);
+
+  for (unsigned int i = 0; i < vel_size_; ++i)
+  {
+    velocity_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < acc_size_; ++i)
+  {
+    acceleration_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < torque_size_; ++i)
+  {
+    joint_torque_[i] = 0.0;
+  }
+
+  // Setup Butterworth filter
+}
+
+InertiaEstimator::InertiaEstimator(hardware_interface::JointHandle joint, double lambda, size_t acc_size)
+{
+  lambda_ = lambda;
+  acc_size_ = acc_size;
+  joint_inertia_ = 0.0;
+
+  // Size the buffers
+  velocity_array_.resize(vel_size_);
+
+  acceleration_array_.resize(acc_size_);
+
+  filtered_acceleration_array_.resize(acc_size_);
+
+  joint_torque_.resize(torque_size_);
+
+  filtered_joint_torque_.resize(fil_tor_size_);
+
+  setJoint(joint);
+
+  for (unsigned int i = 0; i < vel_size_; ++i)
+  {
+    velocity_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < acc_size_; ++i)
+  {
+    acceleration_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < torque_size_; ++i)
+  {
+    joint_torque_[i] = 0.0;
+  }
+
+  // Setup Butterworth filter
+}
+
+InertiaEstimator::~InertiaEstimator()
+{
+}
+
+double InertiaEstimator::getAcceleration(unsigned int index)
+{
+  return acceleration_array_[index];
+}
+
+void InertiaEstimator::setJoint(hardware_interface::JointHandle joint)
+{
+  joint_ = joint;
+}
+
+void InertiaEstimator::setLambda(double lambda)
+{
+  lambda_ = lambda;
+}
+void InertiaEstimator::setAcc_size(size_t acc_size)
+{
+  acc_size_ = acc_size;
+}
+
+void InertiaEstimator::setNodeHandle(ros::NodeHandle& nh)
+{
+  nh_ = nh;
+}
+
+void InertiaEstimator::configurePublisher(std::string name)
+{
+  std::string publisher_name = "/inertia_publisher/" + name;
+  pub_ = nh_.advertise<std_msgs::Float64>(publisher_name, 100);
+}
+
+void InertiaEstimator::publishInertia()
+{
+  std_msgs::Float64 msg;
+  msg.data = joint_inertia_;
+  pub_.publish(msg);
+}
+
+// Fills the buffers so that non-zero values may be computed by the inertia estimator
+bool InertiaEstimator::fill_buffers(double velocity, double effort, const ros::Duration& period)
+{
+  auto it = velocity_array_.begin();
+
+  it = velocity_array_.begin();
+  it = velocity_array_.insert(it, velocity);
+  velocity_array_.resize(vel_size_);
+
+  // Automatically fills the zero'th position of the acceleration array
+  discrete_speed_derivative(velocity, period);
+  acceleration_array_.resize(acc_size_);
+
+  it = joint_torque_.begin();
+  it = joint_torque_.insert(it, effort);
+  joint_torque_.resize(torque_size_);
+  return false;
+}
+
+void InertiaEstimator::apply_butter()
+{
+  // Dingen doen met de sos filter ofzo
+  // Also doe pushback dingen anders dan werkt dit niet
+  std::vector<double> z = { 0.0, 0.0, 0.0 };
+  auto it = filtered_acceleration_array_.begin();
+  auto x_n = acceleration_array_[0];
+  double x;
+
+  for (unsigned int i = 0; i < (sizeof(sos_) / sizeof(sos_[0])); ++i)
```

Comment:
Use range based for loops
```suggestion
  for (const auto& so : sos_)
```